### DISCUSSION TITLE
New Plugin: Filler alt="" attribute

### DISCUSSION THREAD

**Discussion by gpsblues** [GetPublii/Publii#1822] [Useful Resources]
## New Plugin: Filler alt="" attribute  

Hello everyone,  

After a [conversation](https://github.com/GetPublii/Publii/discussions/1808) in the community, I created this plugin to fill empty `alt` attributes in `<img>` tags with the file name. 

👉 [Download **Alt Filler**](https://github.com/gpsblues/Publii-Plugin-Alt-Filler).  

Using regular expressions, the plugin replaces empty `alt` attributes with the file name (without the extension), converting characters like `-`, `.`, and `_` into spaces.  

It works on featured images, individual images, and galleries, making it a **time-saver for production**—especially if file names are already **optimized for SEO**.  

This is a pre-release version, as field testing is still pending. Your feedback is invaluable let me know how it works for you or if there’s room for improvement!  

Happy designing!

[Github repository](https://github.com/gpsblues/Publii-Plugin-Alt-Filler).

**Comment by Vinc26** [comment]
Oh yes thanks!!! I'll try it ASAP!!! 😃

**Comment by Vinc26** [comment]
Well, it works very well!

Just an idea of improvement: to have de possibility to add custom words in plugin settings that to be ignored in the file name. (for sample: "thumbnail")

Thanks so much! It really time saving!

**Reply by gpsblues** [reply]
Hi!
Thanks so much for your feedback :)

I’ve conducted further testing and just released V.1. I slightly improved text formatting, added a final period, and removed the -thumbnail suffix from gallery images.

https://github.com/gpsblues/Publii-Plugin-Alt-Filler
